fabric-data-generation-api-v1/src/main/java/net/fabricmc/fabric/api/datagen/v1/provider/LanguageConsumer.java

```diff
@@ -0,0 +1,146 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.api.datagen.v1.provider;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Map;
+
+import com.google.gson.Gson;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+
+import net.minecraft.block.Block;
+import net.minecraft.enchantment.Enchantment;
+import net.minecraft.entity.EntityType;
+import net.minecraft.entity.attribute.EntityAttribute;
+import net.minecraft.entity.effect.StatusEffect;
+import net.minecraft.item.Item;
+import net.minecraft.item.ItemGroup;
+import net.minecraft.stat.StatType;
+import net.minecraft.util.Identifier;
+
+/**
+ * A consumer used by {@link FabricLanguageProvider#generateLanguages(LanguageConsumer)}.
+ */
+@FunctionalInterface
+public interface LanguageConsumer {
+	/**
+	 * Adds a language entry.
+	 *
+	 * @param languageKey  The key of the language entry.
+	 * @param value        The value of the entry.
+	 */
+	void addLanguage(String languageKey, String value);
```

Comment:
I don't think an entry of a language map is called "language".

Reply:
Just "add" should be fine right?

Reply:
I believe the key is called "translation key" and the value is "translation".